cl/lib/elasticsearch_utils.py

```diff
@@ -2520,13 +2497,16 @@ def limit_inner_hits(
 
 
 def get_child_top_hits_limit(
-    search_params: QueryDict | CleanData, search_type: str
+    search_params: QueryDict | CleanData,
+    search_type: str,
+    api_version: Literal["v3", "v4"] | None = None,
```

Comment:
I believe we can combine the match-case statements. They seem to share similar logic.

Reply:
Sure! I've applied the suggestion, thanks!